### PR TITLE
8282887: Potential memory leak in sun.util.locale.provider.HostLocaleProviderAdapterImpl.getNumberPattern() on Windows

### PR DESCRIPTION
**Pull request by zhengyu123**
Please review this small patch that fixes early `CHECK_NULL_RETURN` results not releasing native `jchar` array returned by `GetStringChars()`.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8282887](https://bugs.openjdk.java.net/browse/JDK-8282887): Potential memory leak in sun.util.locale.provider.HostLocaleProviderAdapterImpl.getNumberPattern() on Windows

### Reviewers
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**)
 * [Alan Bateman](https://openjdk.java.net/census#alanb) (@AlanBateman - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/7777/head:pull/7777` \
`$ git checkout pull/7777`

Update a local copy of the PR: \
`$ git checkout pull/7777` \
`$ git pull https://git.openjdk.java.net/jdk pull/7777/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 7777`

View PR using the GUI difftool: \
`$ git pr show -t 7777`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/7777.diff">https://git.openjdk.java.net/jdk/pull/7777.diff</a>

</details>

### CODE REVIEW

**Review by naotoj — APPROVED** [review]
LGTM.

**Comment by zhengyu123** [comment]
Friendly ping. May I have a second review, please!

**Review by AlanBateman — APPROVED** [review]

**Comment by zhengyu123** [comment]
Thanks, @naotoj @AlanBateman

**Pull request closed without merging**
